freedomofpress/securedrop
# Improve pluralization of translated strings

## Status

Ready for review

## Description of Changes

Switch a few strings containing numeric variables to `ngettext` from `gettext`, to ensure they can be properly translated. Use "num" more consistently as the placeholder for numeric variables in strings marked with `ngettext`, to make things easier for translators.

Fixes #5566.

## Testing

- `git checkout -b more-plurals origin/more-plurals`
- `make translate`
- Choose a test language, e.g. `de_DE` and edit its `.po` file (`securedrop/translations/de_DE/LC_MESSAGES/messages.po`) to find the updated source strings, remove the `fuzzy` markers, and correct the placeholder names.
- In that `.po` file's directory, run `msgfmt messages.po`.
- Run `make dev`.
- Visit the journalist interface, switch to your test language, and ensure that the strings are being translated correctly:
  - Visit the admin page to add a new user. 
    - Enter a HOTP secret a few characters long.
    - Enter less than three characters in the username field.
    - Enter more than 100 characters in the first and last name fields.
    - Pressing add user should result in properly-translated error messages. 
  - Visit the source list and verify that the document, message, and unread counts are being properly translated. You'll need to submit a single file and document as a new source to test singular translations.
  - Delete a single source from the source list. The `{num} collection deleted` message should be properly translated.
  - Delete more than one source from the source list. The `{num} collections deleted` message should be properly translated.
  - Delete a single source submission. The `Submission deleted` flash message should be properly translated.
  - Delete more than one source submission. The `{num} submissions deleted` flash message should be properly translated.

## Deployment

The updated strings will need to be translated at the next release. Until then, they'll always appear in English. 

## Checklist

### If you made changes to the server application code:


### If you made non-trivial code changes:
